src/fe-text/gui-entry.c

```diff
@@ -700,6 +726,9 @@ void gui_entry_erase(GUI_ENTRY_REC *entry, int size, CUTBUFFER_UPDATE_OP update_
 	g_memmove(entry->text + entry->pos - size, entry->text + entry->pos,
 		  (entry->text_len-entry->pos+1) * sizeof(unichar));
 
+	g_memmove(entry->colors + entry->pos - size, entry->colors + entry->pos,
```

Comment:
The `memmove` above has a `+1` that's missing here, why is that?

Reply:
I'm not entirely sure. This is straight from the original patch.

Is `entry->text` `NULL`-terminated, and `entry->colors` implicitly has the same number of elements?